src/processor_arm.cpp

```diff
@@ -1067,31 +1075,55 @@ static CPU get_cpu_name(CPUID cpuid)
-        case 0x28:
-        case 0x29:
+        case 0x22: // H13G Tonga e-Core "Icestorm" used in Apple M1
+        case 0x23: // H13G Tonga p-Core "Firestorm" used in Apple M1
+        case 0x24: // H13J Jade Chop e-Core "Icestorm" used in Apple M1 Pro
+        case 0x25: // H13J Jade Chop p-Core "Firestorm" used in Apple M1 Pro
+        case 0x28: // H13J Jade Die e-Core "Icestorm" used in Apple M1 Max / Ultra
+        case 0x29: // H13J Jade Die p-Core "Firestorm" used in Apple M1 Max / Ultra
             return CPU::apple_m1;
-        case 0x30: // Blizzard m2
-        case 0x31: // Avalanche m2
-        case 0x32:
-        case 0x33:
-        case 0x34:
-        case 0x35:
-        case 0x38:
-        case 0x39:
+        case 0x30: // H14 Ellis e-Core "Blizzard" used in Apple A15
+        case 0x31: // H14 Ellis p-Core "Avalanche" used in Apple A15
+            return CPU::apple_a15;
+        case 0x32: // H14G Staten e-Core "Blizzard" used in Apple M2
+        case 0x33: // H14G Staten p-Core "Avalanche" used in Apple M2
+        case 0x34: // H14S Rhodes Chop e-Core "Blizzard" used in Apple M2 Pro
+        case 0x35: // H14S Rhodes Chop p-Core "Avalanche" used in Apple M2 Pro
+        case 0x38: // H14C Rhodes Die e-Core "Blizzard" used in Apple M2 Max / Ultra
+        case 0x39: // H14C Rhodes Die p-Core "Avalanche" used in Apple M2 Max / Ultra
             return CPU::apple_m2;
-        case 0x49: // Everest m3
-        case 0x48: // Sawtooth m3
+        case 0x40: // H15 Crete e-Core "Sawtooth" used in Apple A16
+        case 0x41: // H15 Crete p-Core "Everest" used in Apple A16
+            return CPU::apple_a16;
+        case 0x42: // H15 Ibiza e-Core "Sawtooth" used in Apple M3
+        case 0x43: // H15 Ibiza p-Core "Everest" used in Apple M3
+        case 0x44: // H15 Lobos e-Core "Sawtooth" used in Apple M3 Pro
+        case 0x45: // H15 Lobos p-Core "Everest" used in Apple M3 Pro
+        case 0x49: // H15 Palma e-Core "Sawtooth" used in Apple M3 Max
+        case 0x48: // H15 Palma p-Core "Everest" used in Apple M3 Max
             return CPU::apple_m3;
+        //case 0x46: // M11 e-Core "Sawtooth" used in Apple S9
+        //case 0x47:  does not exist
+            //return CPU::apple_s9;
+        case 0x50: // H15 Coll e-Core "Sawtooth" used in Apple A17 Pro
+        case 0x51: // H15 Coll p-Core "Everest" used in Apple A17 Pro
+            return CPU::apple_a17;
+        case 0x52: // H16G Donan e-Core used in Apple M4
+        case 0x53: // H16H Donan p-Core used in Apple M4
+        //case 0x??: // H16 Brava e-Core used in Apple M4 Max
+        //case 0x??: // H16 Brava p-Core used in Apple M4 Max
+            return CPU::apple_m4;
+        //case 0x??: // H16 Tahiti e-Core used in Apple A18 Pro
+        //case 0x??: // H16 Tahiti p-Core used in Apple A18 Pro
+        //case 0x??: // H16 Tupai e-Core used in Apple A18
+        //case 0x??: // H16 Tupai p-Core used in Apple A18
```

Comment:
```suggestion
        //case 0x60: // H17P Tahiti e-Core used in Apple A18 Pro
        //case 0x61: // H17P Tahiti p-Core used in Apple A18 Pro
        //case 0x6a: // H17A Tupai e-Core used in Apple A18
        //case 0x6b: // H17A Tupai p-Core used in Apple A18
```

Reply:
`apple-a18` is an alias for apple-m4 in llvm so if you want to add it it shouldn't be too hard.